clmm/theory/parent_class.py

```diff
@@ -371,21 +381,21 @@ def eval_reduced_tangential_shear(self, r_proj, z_cl, z_src, z_src_model='single
         #                               'integrating distribution of redshifts in each radial bin')
         elif z_src_model == 'applegate14':
             if beta_s_mean is None or beta_s_square_mean is None:
-                raise ValueError("beta_s_mean or beta_s_square_mean is not given.")
-            else:
-                z_source = 1000. #np.inf # INF or a very large number
-                gammat = self._eval_tangential_shear(r_proj, z_cl, z_source)
-                kappa = self._eval_convergence(r_proj, z_cl, z_source)
-                gt = beta_s_mean * gammat / (1. - beta_s_square_mean / beta_s_mean * kappa)
+                beta_s_mean = compute_Bs_mean (z_cl, 1000., self.cosmo, pdz=pdz)
+                beta_s_square_mean = compute_Bs_square_mean (z_cl, 1000., self.cosmo, pdz=pdz)
+            z_source = 1000. #np.inf # INF or a very large number
+            gammat = self._eval_tangential_shear(r_proj, z_cl, z_source)
+            kappa = self._eval_convergence(r_proj, z_cl, z_source)
+            gt = beta_s_mean * gammat / (1. - beta_s_square_mean / beta_s_mean * kappa)
         
         elif z_src_model == 'schrabback18':
             if beta_s_mean is None or beta_s_square_mean is None:
-                raise ValueError("beta_s_mean or beta_s_square_mean is not given.")
-            else:
-                z_source = 1000. #np.inf # INF or a very large number
-                gammat = self._eval_tangential_shear(r_proj, z_cl, z_source)
-                kappa = self._eval_convergence(r_proj, z_cl, z_source)
-                gt = (1. + (beta_s_square_mean / (beta_s_mean * beta_s_mean) - 1.) * beta_s_mean * kappa) * (beta_s_mean * gammat / (1. - beta_s_mean * kappa))
+                beta_s_mean = compute_Bs_mean (z_cl, 1000., self.cosmo, pdz=pdz)
+                beta_s_square_mean = compute_Bs_square_mean (z_cl, 1000., self.cosmo, pdz=pdz)
+            z_source = 1000. #np.inf # INF or a very large number
+            gammat = self._eval_tangential_shear(r_proj, z_cl, z_source)
```

Comment:
Mention z_source beforehand so we can have z_inf =  z_source in the functions instead of hardcoded 1000.